app/controllers/two_factor_authentication/backup_code_verification_controller.rb

```diff
@@ -19,7 +19,9 @@ def show
     def create
       @backup_code_form = BackupCodeVerificationForm.new(current_user)
       result = @backup_code_form.submit(backup_code_params)
-      analytics.track_mfa_submit_event(result.to_h)
+      analytics_result = result.to_h
+      analytics_result[:new_device] = current_user.new_device(cookies)
+      analytics.track_mfa_submit_event(analytics_result)
```

Comment:
my vote is to just inline things like this
```suggestion
      analytics.track_mfa_submit_event(result.to_h.merge(new_device: current_user.new_device(cookies))
```

Reply:
I agree. Thanks!